packages/GA4Client/src/common.js

```diff
@@ -43,7 +45,8 @@ Common.prototype.truncateAttributes = function (
 
     if (!isEmpty(attributes)) {
         Object.keys(attributes).forEach(function (attribute) {
-            var key = truncateString(attribute, keyLimit);
+            var standardizedKey = attribute.replace(/[^a-zA-Z0-9_]/g, '_');
```

Comment:
Since you are using this multiple times throughout, I would move this out of the functions and make it a variable at the top of the file so it can be reused.  

Something like:
```
var alphabetRegex = /[^a-zA-Z0-9_]/g;
var standardizedKey = attribute.replace(alphabetRegex, '_')
```